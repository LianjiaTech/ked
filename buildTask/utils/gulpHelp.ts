import path from 'path';

const cwd = process.cwd();

function getProjectPath(...filePath: string[]) {
  return path.join(cwd, ...filePath);
}

function resolve(moduleName: string) {
  return require.resolve(moduleName);
}

export { getProjectPath, resolve };
