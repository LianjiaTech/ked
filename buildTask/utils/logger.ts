import chalk from 'chalk';
import { EOL } from 'os';
import { format } from 'util';

const colors = {
  log: (text) => text,
  info: chalk.green.bind(chalk),
  warn: chalk.yellow.bind(chalk),
  error: chalk.red.bind(chalk),
};

function wrapConsole(originConsole) {
  const newConsole = Object.create(originConsole);
  ['log', 'info', 'warn', 'error'].forEach(name => {
    const func = newConsole[name];
    newConsole[name] = (formater, ...args) => {
      const text = format(formater, ...args);
      if (text.includes(EOL)) {
        return func.call(newConsole, formater, ...args);
      }
      const time = `[${new Date().toLocaleString()}]`;
      return func.call(newConsole, chalk.blue(time), colors[name](text));
    };
  });
  return newConsole;
}

export const logger = wrapConsole(console);
