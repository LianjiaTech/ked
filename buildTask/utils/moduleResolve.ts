/**
 * @description
 * @date  2016/11/17.
 */
export default (name, nodePath = '') => {
  nodePath &&
    Array.isArray(nodePath) &&
    nodePath.map(path => {
      if (!module.paths.includes(path)) {
        module.paths = module.paths.concat(path);
      }
    });
  return require.resolve(name);
};
