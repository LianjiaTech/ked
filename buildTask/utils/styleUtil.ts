// We convert less import in es/lib to css file path
export function cssInjection(content) {
  return content
    .replace(/\/style\/?'/g, "/style/css'")
    .replace(/\/style\/?"/g, '/style/css"')
    .replace(/\.less/g, '.css');
}
