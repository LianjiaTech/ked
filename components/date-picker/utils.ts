import * as moment from 'moment';

// eslint-disable-next-line import/prefer-default-export
export function formatDate(
  value: moment.Moment | undefined | null,
  format: string | string[] | undefined,
): string {
  if (!value) {
    return '';
  }
  if (Array.isArray(format)) {
    format = format[0];
  }
  return value.format(format);
}
