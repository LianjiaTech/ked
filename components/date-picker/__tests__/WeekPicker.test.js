import React from 'react';
import { mount, render } from 'enzyme';
import moment from 'moment';
import { setMockDate, resetMockDate } from '../../../../tests/utils';
import DatePicker from '..';
import focusTest from '../../../../tests/shared/focusTest';
import { openPanel } from './utils';

const { WeekPicker } = DatePicker;

describe('WeekPicker', () => {
  beforeEach(() => {
    setMockDate();
  });

  afterEach(() => {
    resetMockDate();
  });

  focusTest(WeekPicker);

  it('should support style prop', () => {
    const wrapper = mount(<WeekPicker style={{ width: 400 }} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('extra footer works', () => {
    const wrapper = mount(
      <WeekPicker renderExtraFooter={mode => <span className="extra-node">{mode}</span>} />,
    );
    openPanel(wrapper);

    let extraNode = wrapper.find('.extra-node');
    expect(extraNode.length).toBe(1);
    expect(extraNode.text()).toBe('date');

    wrapper
      .find('.ant-calendar-month-select')
      .hostNodes()
      .simulate('click');
    extraNode = wrapper.find('.ant-calendar-month-panel .extra-node');
    expect(extraNode.length).toBe(1);
    expect(extraNode.text()).toBe('month');

    wrapper
      .find('.ant-calendar-year-select')
      .hostNodes()
      .simulate('click');
    extraNode = wrapper.find('.ant-calendar-year-panel .extra-node');
    expect(extraNode.length).toBe(1);
    expect(extraNode.text()).toBe('year');

    wrapper
      .find('.ant-calendar-year-panel-decade-select')
      .hostNodes()
      .simulate('click');
    extraNode = wrapper.find('.ant-calendar-decade-panel .extra-node');
    expect(extraNode.length).toBe(1);
    expect(extraNode.text()).toBe('decade');
  });

  it('should support dateRender', () => {
    const wrapper = mount(
      <WeekPicker open dateRender={current => <span>{current.format('YYYY-MM-DD')}</span>} />,
    );
    expect(
      render(
        wrapper
          .find('Trigger')
          .instance()
          .getComponent(),
      ),
    ).toMatchSnapshot();
  });

  it('should support allowClear', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <WeekPicker defaultValue={moment()} onChange={onChange} />,
    );
    wrapper
      .find('.ant-calendar-picker-clear')
      .hostNodes()
      .simulate('click');
    expect(onChange).toHaveBeenCalledWith(null, '');
  });
});
