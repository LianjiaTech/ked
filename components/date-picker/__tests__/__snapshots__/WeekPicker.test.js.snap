// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekPicker should support dateRender 1`] = `
<div>
  <div
    class="ant-calendar-picker-container  ant-calendar-picker-container-placement-bottomLeft "
  >
    <div
      class="ant-calendar ant-calendar-week-number"
      tabindex="0"
    >
      <div
        class="ant-calendar-panel"
      >
        <div
          class="ant-calendar-date-panel"
          tabindex="0"
        >
          <div
            class="ant-calendar-header"
          >
            <div
              style="position:relative"
            >
              <a
                class="ant-calendar-prev-year-btn"
                role="button"
                title="Last year (Control + left)"
              />
              <a
                class="ant-calendar-prev-month-btn"
                role="button"
                title="Previous month (PageUp)"
              />
              <span
                class="ant-calendar-my-select"
              >
                <a
                  class="ant-calendar-month-select"
                  role="button"
                  title="Choose a month"
                >
                  Sep
                </a>
                <a
                  class="ant-calendar-year-select"
                  role="button"
                  title="Choose a year"
                >
                  2017
                </a>
              </span>
              <a
                class="ant-calendar-next-month-btn"
                title="Next month (PageDown)"
              />
              <a
                class="ant-calendar-next-year-btn"
                title="Next year (Control + right)"
              />
            </div>
          </div>
          <div
            class="ant-calendar-body"
          >
            <table
              cellspacing="0"
              class="ant-calendar-table"
              role="grid"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    class="ant-calendar-column-header ant-calendar-week-number-header"
                    role="columnheader"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      x
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="Sun"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      Su
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="Mon"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      Mo
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="Tue"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      Tu
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="Wed"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      We
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="Thu"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      Th
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="Fri"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      Fr
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="Sat"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      Sa
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-calendar-tbody"
              >
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    35
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="August 27, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-08-27
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="August 28, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-08-28
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="August 29, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-08-29
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="August 30, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-08-30
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell ant-calendar-last-day-of-month"
                    role="gridcell"
                    title="August 31, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-08-31
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 1, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-01
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 2, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-02
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    36
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 3, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-03
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 4, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-04
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 5, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-05
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 6, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-06
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 7, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-07
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 8, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-08
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 9, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-09
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    37
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 10, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-10
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 11, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-11
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 12, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-12
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 13, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-13
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 14, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-14
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 15, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-15
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 16, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-16
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-calendar-current-week ant-calendar-active-week"
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    38
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 17, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-17
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-today ant-calendar-selected-day"
                    role="gridcell"
                    title="September 18, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-18
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 19, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-19
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 20, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-20
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 21, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-21
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 22, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-22
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 23, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-23
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    39
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 24, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-24
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 25, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-25
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 26, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-26
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 27, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-27
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 28, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-28
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="September 29, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-29
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-day-of-month"
                    role="gridcell"
                    title="September 30, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-09-30
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    40
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="October 1, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-10-01
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="October 2, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-10-02
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="October 3, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-10-03
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="October 4, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-10-04
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="October 5, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-10-05
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="October 6, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-10-06
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="October 7, 2017"
                  >
                    <div
                      class="ant-calendar-date"
                    >
                      <span>
                        2017-10-07
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WeekPicker should support style prop 1`] = `
<span
  class="ant-calendar-picker"
  style="width: 400px;"
>
  <span
    style="display: inline-block; width: 100%;"
  >
    <input
      class="ant-calendar-picker-input ant-input"
      placeholder="Select date"
      readonly=""
      value=""
    />
    <i
      aria-label="icon: calendar"
      class="anticon anticon-calendar ant-calendar-picker-icon"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="calendar"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
        />
      </svg>
    </i>
  </span>
</span>
`;
